src/cls/cas/cls_cas.cc

```diff
@@ -116,8 +116,9 @@ static int cls_rc_chunk_refcount_put(cls_method_context_t hctx, bufferlist *in,
     return 0;
   }
 
-  auto p = objr.refs.find(op.source);
-  objr.refs.erase(p);
+  auto p = std::find(objr.refs.begin(), objr.refs.end(), op.source);
```

Comment:
This would appear to duplicate the logic just a few lines up.  Can you remove the other search?

Reply:
I'm worried about replacing the log(n) std::set search with a linear vector scan.  Is there a reason you didn't choose to use something like std::multiset?